lib/crypto/secp256k1/secp256k1.go

```diff
@@ -48,11 +51,27 @@ type PublicKey struct {
 
 // RecoverPublicKey returns the 64-byte uncompressed public key that created the given signature.
 func RecoverPublicKey(msg, sig []byte) ([]byte, error) {
+	if sig[64] == 27 {
+		sig[64] = 0
+	}
+
+	if sig[64] == 28 {
```

Comment:
Not needed but you can combine both conditions if it makes sense.
```
	if sig[64] == 27 || sig[64] == 28  {
		sig[64] -= 27
	}
````
or as done here
https://github.com/paritytech/substrate/blob/2ab5cd2880b1ec798a7c2a72c29f995f54b45f13/primitives/io/src/lib.rs#L767
```
        const someVal = 27
	if sig[64] >= someVal {
		sig[64] -= someVal
	}
````

Reply:
done!

Reply:
@noot Same can be done for `RecoverPublicKeyCompressed`

Reply:
also done!